src/core/Akka/Dispatch/Mailbox.cs

```diff
@@ -338,10 +338,12 @@ private void ProcessAllSystemMessages()
                 msg.Unlink();
                 DebugPrint("{0} processing system message {1} with {2}", Actor.Self, msg, string.Join(",", Actor.GetChildren()));
                 // we know here that SystemInvoke ensures that only "fatal" exceptions get rethrown
+#if UNSAFE_THREADING
```

Comment:
As far as I know the ThreadInterruptedException and ThreadAbortException handlers here are useless.

Actor.SystemInvoke isn't blocking, so ThreadInterruptedException can't be thrown.

In the ThreadAbortException handler, Thread.ResetAbort() isn't called so the exception will be [re-thrown automatically](https://msdn.microsoft.com/en-us/library/system.threading.threadabortexception(v=vs.110).aspx#Anchor_4).

ThreadAbort resistance is very hard to get right and I don't think it is properly implemented here even if Thread.ResetAbort was called, I would advise to simply remove thoses exceptions handlers and never abort threads.